### PR TITLE
NO-JIRA: Several fixes for Keycloack IDP test

### PR DESCRIPTION
**Pull request by vrutkovs**
* run deployment instead of a pod
* ping Keycloack version to v25
* since Keycloak v25 `/health` is no longer exposed on public ports. Use deployment probes instead

### CODE REVIEW

**Review by liouk — CHANGES_REQUESTED** [review]

File: test/library/idpdeployment.go [hunk 1]
```diff
@@ -131,6 +136,21 @@ func deployPod(
 	_, err = clients.CoreV1().Services(namespace).Create(testContext, svcTemplate(httpPort, httpsPort), metav1.CreateOptions{})
 	require.NoError(t, err)
 
+	timeLimitedCtx, cancel := context.WithTimeout(testContext, 5*time.Minute)
+	defer cancel()
+	if _, err := watchtools.UntilWithSync(timeLimitedCtx,
```
[the remark this answers is not in the document]
Comment (vrutkovs) [reply test/library/idpdeployment.go:141]: I think `deployPod` should do the basic wait (image pulling, container creation etc.) and other parts may want to add separate waits if necessary

**Comment by liouk** [comment]
Re 41b65b49e75a4cc8752c2defd375166c36a8eaec, I'll make sure we keep an eye on Keycloak releases and update the test as necessary. Breaking tests would force us to get them fixed and adapt to the latest Keycloak version, but that's not necessary in this context; so better be safe than sorry :) 

/lgtm

**Review by soltysh — APPROVED** [review]
/label approved

**Pull request merged**